gridded_etl_tools/dataset_manager.py

```diff
@@ -297,6 +302,38 @@ def transform(self):
         # from Dask
         self.create_zarr_json()
 
+    def transform_dataset_in_memory(self) -> xr.Dataset:
```

Comment:
I'm actually concerned about this function returning a value now that I look at managers. Previously we just transformed the ETL object internally/in-place. 

I think returning values is one way to make ETL objects more versatile, but the advantage of not returning the dataset here was a client using a pre-made object like CPCPrecipUS did not need to know about xarray.

I think in general we want to preserve a concept of a manager which is able to run an ETL just by running functions. And especially in this case it seems counter-intuitive to give the client an unfinished dataset and expect the client to do the next step with it.

Reply:
@eschechter could you chime in here, since we very specifically spoke about this?